test_conformance/c11_atomics/test_atomics.cpp

```diff
@@ -230,9 +230,11 @@ class CBasicTestLoad : public CBasicTestMemOrderScope<HostAtomicType, HostDataTy
   {
     std::string memoryOrderScope = MemoryOrderScopeStr();
     std::string postfix(memoryOrderScope.empty() ? "" : "_explicit");
-    return
-      "  atomic_store(&destMemory[tid], tid);\n"
-      "  oldValues[tid] = atomic_load"+postfix+"(&destMemory[tid]"+memoryOrderScope+");\n";
+    return "  atomic_store" + postfix + "(&destMemory[tid], tid"
+        + memoryOrderScope
```

Comment:
I suspect that this is capable of passing `memory_order_acquire` to `atomic_store_explicit`?  If so that is illegal.

If the test is passing `memory_order_acquire` to a `atomic_load_explicit` on the next line then this line should pass `memory_order_release` to the `atomic_store_explicit`.

This will only affect implementations that report `CL_DEVICE_ATOMIC_ORDER_ACQ_REL` (which is likely why it worked for you but is not correct in general).

Reply:
Ah good point, before it was always `memory_order_seq_cst`, I'll add some logic to handle these cases.

Reply:
I've updated this to use `memory_order_release` if the load uses `atomic_order_acquire`. Let me know if this is what you had in mind. It would be super useful if someone could test this on an implementation that supports these orders and confirm that this behaves as expected, unfortunately I can't do this locally.

Reply:
Yes, this is what I had in mind.